### DISCUSSION TITLE
Restart server if it crashes (and start from batch file)

### DISCUSSION THREAD

**Discussion by coragin** [Ch4r0ne/ARK-Ascended-Server-Manager#17] [Ideas]
Hi,
New to GitHub, not to running ARK servers (been doing ASE for years and years).  Here is my question/idea.
I have always ran servers from a batch file, something that gives it instructions on where to custom save the server, cluster rules, etc.  Looks something like this (This is ASA batch file to run the island with a LOT of mods.

RunIslandServer.bat
`start ArkAscendedServer.exe "TheIsland_WP?Port=7777?QueryPort=27015?SessionName=[NA][Raging Dragon]Lots-Deco|+Taming+Breed+Stack,+Harvest,+XP?ServerAdminPassword=xxxxxxxx?AltSaveDirectoryName=@2023-10-30The_Island?MaxPlayers=70?listen?OverrideStructurePlatformPrevention=true?AllowCaveBuildingPvE=True?NonPermanentDiseases=true?" -mods=935676,928793,930561,929506,929521,929543,929578,929745,929775,929977,930025,930361,930436,930829,930494,928824,932221,931558,932218,933158,931877,930115,928825,928539,933447,933731,933489,934365,933638,930391,934384,928664,930359,934401,935617,935715,935204,933588,933567,929299,928862`

With the various crashes that are happening, is there a way to run ASA server bat file instead of the "ArkAscendedServer.exe" to run "RunIslandServer.bat"?  If not, this is idea one which leads into idea two...

Is it possible to have the server manager "watch" or "check" that the ArkAscendedServer.exe is actually running?  And if its not (like from a crash), re-run the server from the same bat file?

Finally a question. Is there a way to have something 'check' for ArkAscendedServer.exe (like in task manager) and if its not running, to run 'RunIslandServer.bat'?

I love what you have laid out here and its nice to have a program to all the things yours does, including the options and option to open GameUserSettings.ini and Game.ini with just a single click rather than have to open directories or place shortcuts etc and have a lot of windows open.

Mainly what I would LOVE is for the ability to do what you have already done, but also and mostly, run that batch file if the server stops running for whatever reason.  In my case, something is randomly crashing my server and I have yet to track it down, if its a mod, resource collection is too high on something and its causing a server crash, Im not sure.  Okay, I am going to stop rambling here and hope people will reply lol

**Comment by Ch4r0ne** [comment]
@coragin This skirpt should solve your problem if you have already submitted an AutoUpdateJob so that the .ps1 and the config file matches the server manager.

This creates a task in the tasksceduler that checks whether the server is running, if it is not running, then the .ps1 is started, this checks for updates and also restarts the server based on the set configuration.

Create an .ps1 or use the PowerShellISE and run the skript as Administrator:

```
# Set the path of the file to be executed
$psScriptPath = "$env:APPDATA\ARK-Ascended-Server-Manager\AutoUpdateJob.ps1"

# Set the interval in minutes
$interval = 10

# Set the task name
$taskName = "AutoRestartASAServer"

# Create task planning with Powershell Command
schtasks /create /tn $taskName /tr "powershell.exe -ExecutionPolicy Bypass -Command 'if (-not (Get-Process -Name ArkAscendedServer -ErrorAction SilentlyContinue)) { Start-Process -FilePath $psScriptPath -PassThru}'" /sc minute /mo $interval /ru SYSTEM
```

Please note that I have not yet tested this and that small adjustments may still be necessary if it does not recognise the process.

**Reply by Ch4r0ne** [reply]
In this skirpt, you can of course also make slight changes and also execute a .bat.
